application/constants/templates.py

```diff
@@ -43,3 +53,62 @@ class HookTypes(StrEnum):
     Template application hook types.
     """
     kubernetes_job = 'kubernetes_job'
+
+
+class ComponentTypeHttpMethods(StrEnum):
+    """
+    Template HTTP component methods.
+    """
+    get = 'get'
+    post = 'post'
+    put = 'put'
+    patch = 'patch'
+    delete = 'delete'
+
+
+class HttpDeploy(BaseModel):
```

Comment:
I just noticed that you have added Pydantic HTTP models to constants file. This is not proper place to define it. On my opinion it should be placed in `application/schemas/templates/components.py`.

Reply:
*After chat in telegram*

I think it should one Pydantic model but(parameter for all of them are identical) you need to add parameter to distinguish it for (un)install or status. Something like `trigger: ['install' | 'uninstall' | 'status']`.